space_packet_parser/parser.py

```diff
@@ -445,24 +445,27 @@ def fill_read_buffer(source: bitstring.ConstBitStream or BinaryIO or socket.sock
                 Number of bits added to the buffer. Note that the buffer may still have nonzero length from previous
                 data even when this returns zero.
             """
+            curser_pos = buffer.pos  # Keep track of the original buffer cursor location
```

Comment:
```suggestion
            cursor_pos = buffer.pos  # Keep track of the original buffer cursor location
```

Reply:
Yeah, I understand why the Bitstring author made the change. This just makes my code behave the same way the old Bitstring worked because we fundamentally need the ringbuffer cursor not to move just because we added some data to it.